.github/workflows/hygiene.yml

```diff
@@ -0,0 +1,44 @@
+on: [push, pull_request]
+
+name: "Code Hygiene Suite"
+
+env:
+    CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
+
+jobs:
+  hygiene:
+    strategy:
+      matrix:
+        platform: [macos-latest, ubuntu-20.04, windows-latest]
```

Comment:
I'm used to only running static checks on one platform, it's easier to maintain.

Reply:
Happy to reduce it if you want.

- `fmt` won't (shouldn't) have any OS quirks.
- `clippy` and `check` will reify dependencies and [platform specific dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies) are a thing that could fail on one OS but not another, but in practice, I've never seen it get triggered via these two because I usually only am building in Linux.

If left in, I'd probably only leave it in for `cargo check`.

Thoughts, @lukipuki ?

Reply:
You are right, there could be platform specific dependencies and code, let's keep it then.